### ISSUE TITLE
ES6: when exploding an array, eslint complaints of unused

### ISSUE THREAD

**Issue by gre** [eslint/eslint#3288]
``` 
const [ x, y, w, h, pageX, pageY ] = measure;
f(pageY);
```

`* is defined but never used`  error will appear for `x, y, w, h, pageX`, but it should not because I'm using the latest pageY

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by mysticatea** [comment]
Thank you for the report.

I thought this is an correct behavior.
You can use one of below:

```js
const [ , , , , , pageY ] = measure;
```

```js
const pageY = measure[5];
```

```js
const [ x, y, w, h, pageX, pageY ] = measure; // eslint-disable-line no-unused-vars
```

However, `const [ x, y, w, h, pageX, pageY ] = measure;` is self-described and readable, so I will not oppose supporting this.

**Comment by michaelficarra** [comment]
This issue has been raised before, and I believe the consensus was to use holes, as @mysticatea shows.

**Comment by gre** [comment]
Oh yeah, I've forgotten about this es6 feature. Make sense, thanks
